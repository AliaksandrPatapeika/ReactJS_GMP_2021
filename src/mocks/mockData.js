import image01 from '../img/posters/01-pulp-fiction.jpg';
// import image02 from '../img/posters/02-bohemian-rhapsody.jpg';
import image03 from '../img/posters/03-kill-bill.jpg';
import image04 from '../img/posters/04-avengers-infinity-war.jpg';
import image05 from '../img/posters/05-inception.jpg';
import image06 from '../img/posters/06-reservoir-dogs.jpg';

export const MockMovies = [
  {
    id: 1,
    movieURL: image01,
    name: 'Pulp Fiction',
    releaseDate: '2004-03-15',
    genre: 'Action & Adventure',
    runtime: 154,
    rating: 4.3,
    overview: 'Pulp Fiction is a 1994 American neo-noir black comedy crime film written and directed by Quentin Tarantino, who conceived it with Roger Avary.[4] Starring John Travolta, Samuel L. Jackson, Bruce Willis, Tim Roth, Ving Rhames, and Uma Thurman, it tells several stories of criminal Los Angeles. The title refers to the pulp magazines and hardboiled crime novels popular during the mid-20th century, known for their graphic violence and punchy dialogue.'
  },
  {
    id: 2,
    // movieURL: image02,
    name: 'Bohemian Rhapsody',
    releaseDate: '2003-05-23',
    genre: 'Drama, Biography, Music'
  },
  {
    id: 3,
    movieURL: image03,
    name: 'Kill Bill: Vol 2',
    releaseDate: '1994-01-08',
    genre: 'Oscar winning Movie',
    runtime: 122,
    rating: 4.7,
    overview: 'Kill Bill: Volume 2 is a 2004 American martial arts film written and directed by Quentin Tarantino. It stars Uma Thurman as the Bride, who continues her campaign of revenge against the Deadly Viper Assassination Squad (Lucy Liu, Michael Madsen, Daryl Hannah, and Vivica A. Fox) and their leader Bill (David Carradine), who tried to kill her and her unborn child.'
  },
  {
    id: 4,
    movieURL: image04,
    name: 'Avengers: War of Infinity',
    releaseDate: '2004-07-21',
    genre: 'Action & Adventure',
    runtime: 98,
    rating: 4.2,
    overview: 'Avengers: Infinity War is a 2018 American superhero film based on the Marvel Comics superhero team the Avengers. Produced by Marvel Studios and distributed by Walt Disney Studios Motion Pictures, it is the sequel to The Avengers (2012) and Avengers: Age of Ultron (2015), and the 19th film in the Marvel Cinematic Universe (MCU). Directed by Anthony and Joe Russo and written by Christopher Markus and Stephen McFeely, the film features an ensemble cast including Robert Downey Jr., Chris Hemsworth, Mark Ruffalo, Chris Evans, Scarlett Johansson, Benedict Cumberbatch, Don Cheadle, Tom Holland, Chadwick Boseman, Paul Bettany, Elizabeth Olsen, Anthony Mackie, Sebastian Stan, Danai Gurira, Letitia Wright, Dave Bautista, Zoe Saldana, Josh Brolin, and Chris Pratt. In the film, the Avengers and the Guardians of the Galaxy attempt to prevent Thanos from collecting the six all-powerful Infinity Stones as part of his quest to kill half of all life in the universe.'
  },
  {
    id: 5,
    movieURL: image05,
    name: 'Inception',
    releaseDate: '2003-05-02',
    genre: 'Action & Adventure',
    runtime: 105,
    rating: 4.5,
    overview: 'Inception is a 2010 science fiction action film[4][5][6] written and directed by Christopher Nolan, who also produced the film with his wife, Emma Thomas. The film stars Leonardo DiCaprio as a professional thief who steals information by infiltrating the subconscious of his targets. He is offered a chance to have his criminal history erased as payment for the implantation of another person\'s idea into a target\'s subconscious.[7] The ensemble cast includes Ken Watanabe, Joseph Gordon-Levitt, Marion Cotillard, Elliot Page,[a] Tom Hardy, Dileep Rao, Cillian Murphy, Tom Berenger, and Michael Caine.'
  },
  {
    id: 6,
    movieURL: image06,
    name: 'Reservoir dogs',
    releaseDate: '1994-11-30',
    genre: 'Oscar winning Movie',
    runtime: 113,
    rating: 4.1,
    overview: 'Reservoir Dogs is a 1992 American crime film written and directed by Quentin Tarantino in his feature-length debut. It stars Harvey Keitel, Tim Roth, Chris Penn, Steve Buscemi, Lawrence Tierney, Michael Madsen, Tarantino, and Edward Bunker as diamond thieves whose planned heist of a jewelry store goes terribly wrong. The film depicts the events before and after the heist. Kirk Baltz, Randy Brooks, and Steven Wright also play supporting roles. It incorporates many motifs that have become Tarantino\'s hallmarks: violent crime, pop culture references, profanity, and nonlinear storytelling.'
  }
];

export const MockResultCount = 39; // 0 for testing No Movies found page

// eslint-disable-next-line no-console
export const MockOnClick = (button, attrs) => console.log(`Clicked from "${button}"`, attrs);
