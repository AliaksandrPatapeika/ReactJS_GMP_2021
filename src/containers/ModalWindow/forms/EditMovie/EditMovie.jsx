import '../../ModalWindow.less';

import PropTypes from 'prop-types';
import React, {useContext, useState} from 'react';

import Button from '../../../../components/Button';
import Select from '../../../../components/Select';
import {Genres} from '../../../../constants';
import MovieContext from '../../../../context';
import {MockOnClick} from '../../../../mocks/mockData';
import getGenres from '../../../../utils';

const EditMovie = ({
  formTitle,
  movie: {
    genres, id, title, release_date, poster_path, overview, runtime
  }
}) => {
  const [optionSelected, setOptionSelected] = useState(getGenres(Genres, genres));

  const selectChange = (selected) => {
    setOptionSelected(selected);
  };

  const selectReset = () => {
    setOptionSelected(getGenres(Genres, genres));
  };

  const {activeModalWindow} = useContext(MovieContext);

  return (
    <>
      {activeModalWindow === 'editMovie' && (
      <>
        <span className="title">{formTitle}</span>
        <form
          onSubmit={() => MockOnClick('form Edit movie onSubmit movie.id', id)}
          onReset={selectReset}
        >
          <div className="formInput">
            <label htmlFor="id" className="label">
              MOVIE ID
            </label>
            <input
              type="text"
              name="id"
              id="id"
              className="input"
              value={id}
              readOnly
            />
          </div>
          <div className="formInput">
            <label htmlFor="title" className="label">
              TITLE
            </label>
            <input
              type="text"
              name="title"
              id="title"
              className="input"
              placeholder="Title here"
              defaultValue={title}
              autoComplete="off"
            />
          </div>
          <div className="formInput">
            <label htmlFor="releaseDate" className="label">
              RELEASE DATE
            </label>
            <input
              type="date"
              name="releaseDate"
              id="releaseDate"
              className="input"
              placeholder="Select Date"
              defaultValue={release_date}
              required
            />
          </div>
          <div className="formInput">
            <label htmlFor="movieURL" className="label">
              MOVIE URL
            </label>
            <input
              type="text"
              name="movieURL"
              id="movieURL"
              className="input"
              placeholder="Movie URL here"
              defaultValue={poster_path}
              autoComplete="off"
            />
          </div>
          <div className="formInput">
            <label htmlFor="genres" className="label">
              GENRE
            </label>
            <Select
              className="select"
              classNamePrefix="select"
              id="genres"
              options={Genres}
              placeholder="Select Genre"
              isMulti
              closeMenuOnSelect={false}
              hideSelectedOptions={false}
              onChange={selectChange}
              allowSelectAll
              allSelectedLabel="All genres"
              value={optionSelected}
            />
          </div>
          <div className="formInput">
            <label htmlFor="overview" className="label">
              OVERVIEW
            </label>
            <input
              type="text"
              name="overview"
              id="overview"
              className="input"
              placeholder="Overview here"
              defaultValue={overview}
              autoComplete="off"
            />
          </div>
          <div className="formInput">
            <label htmlFor="runtime" className="label">
              RUNTIME
            </label>
            <input
              type="text"
              name="runtime"
              id="runtime"
              className="input"
              placeholder="Runtime here"
              defaultValue={runtime}
              autoComplete="off"
            />
          </div>
          <div className="modalFooter">
            <Button
              type="reset"
              className="button"
              title="RESET"
            />
            <Button
              type="submit"
              className="button"
              title="SAVE"
            />
          </div>
        </form>
      </>
      )}
    </>
  );
};

EditMovie.propTypes = {
  formTitle: PropTypes.string.isRequired,
  movie: PropTypes.shape({
    budget: PropTypes.number,
    genres: PropTypes.arrayOf(PropTypes.string),
    id: PropTypes.number,
    overview: PropTypes.string,
    poster_path: PropTypes.string,
    release_date: PropTypes.string,
    revenue: PropTypes.number,
    runtime: PropTypes.number,
    tagline: PropTypes.string,
    title: PropTypes.string,
    vote_average: PropTypes.number,
    vote_count: PropTypes.number
  })
};

EditMovie.defaultProps = {
  movie: {
    genres: []
  }
};

export default EditMovie;
