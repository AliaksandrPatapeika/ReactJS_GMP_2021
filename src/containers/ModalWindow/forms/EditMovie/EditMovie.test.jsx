import {head} from 'lodash';
import React from 'react';
import renderer from 'react-test-renderer';

import MovieContext from '../../../../context';
import {MockMovies} from '../../../../tests/mocks/mockData';
import EditMovie from './EditMovie';

describe('EditMovie', () => {
  test('renders correctly', () => {
    const mockMovie = head(MockMovies);
    const editMovie = renderer
      .create(
        <MovieContext.Provider value={{
				  activeModalWindow: 'editMovie'
        }}
        >
          <EditMovie formTitle="title" movie={mockMovie} />
        </MovieContext.Provider>
      )
      .toJSON();

    expect(editMovie).toMatchSnapshot();
  });
});
