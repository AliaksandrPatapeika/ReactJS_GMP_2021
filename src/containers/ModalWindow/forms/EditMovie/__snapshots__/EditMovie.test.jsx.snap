// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditMovie renders correctly 1`] = `
Array [
  <span
    className="title"
  >
    title
  </span>,
  <form
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="formInput"
    >
      <label
        className="label"
        htmlFor="id"
      >
        MOVIE ID
      </label>
      <input
        className="input"
        id="id"
        name="id"
        readOnly={true}
        type="text"
        value={1}
      />
    </div>
    <div
      className="formInput"
    >
      <label
        className="label required-label"
        htmlFor="title"
      >
        TITLE
      </label>
      <input
        autoComplete="off"
        className="input"
        defaultValue="Pulp Fiction"
        id="title"
        name="title"
        onChange={[Function]}
        placeholder="Title here"
        required={true}
        type="text"
      />
    </div>
    <div
      className="formInput"
    >
      <label
        className="label required-label"
        htmlFor="release_date"
      >
        RELEASE DATE
      </label>
      <input
        className="input"
        defaultValue="2004-03-15"
        id="release_date"
        name="release_date"
        onChange={[Function]}
        placeholder="Select Date"
        required={true}
        type="date"
      />
    </div>
    <div
      className="formInput"
    >
      <label
        className="label"
        htmlFor="poster_path"
      >
        MOVIE URL
      </label>
      <input
        autoComplete="off"
        className="input"
        defaultValue="test-file-stub"
        id="poster_path"
        name="poster_path"
        onChange={[Function]}
        placeholder="Movie URL here"
        type="url"
      />
    </div>
    <div
      className="formInput"
    >
      <label
        className="label required-label"
        htmlFor="genres"
      >
        GENRE
      </label>
      <div
        className="requiredSelectContainer"
      >
        <div
          className="select css-2b097c-container"
          id="genres"
          onKeyDown={[Function]}
        >
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            className="css-1f43avz-a11yText-A11yText"
          />
          <div
            className="select__control css-yk16xz-control"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <div
              className="select__value-container select__value-container--is-multi css-g1d714-ValueContainer"
            >
              <div
                className="select__placeholder css-1wa3eu0-placeholder"
              >
                Select Genre
              </div>
              <div
                className="css-b8ldur-Input"
              >
                <div
                  className="select__input"
                  style={
                    Object {
                      "display": "inline-block",
                    }
                  }
                >
                  <input
                    aria-autocomplete="list"
                    autoCapitalize="none"
                    autoComplete="off"
                    autoCorrect="off"
                    disabled={false}
                    id="react-select-2-input"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    spellCheck="false"
                    style={
                      Object {
                        "background": 0,
                        "border": 0,
                        "boxSizing": "content-box",
                        "color": "inherit",
                        "fontSize": "inherit",
                        "label": "input",
                        "opacity": 1,
                        "outline": 0,
                        "padding": 0,
                        "width": "1px",
                      }
                    }
                    tabIndex="0"
                    type="text"
                    value=""
                  />
                  <div
                    style={
                      Object {
                        "height": 0,
                        "left": 0,
                        "overflow": "scroll",
                        "position": "absolute",
                        "top": 0,
                        "visibility": "hidden",
                        "whiteSpace": "pre",
                      }
                    }
                  >
                    
                  </div>
                </div>
              </div>
            </div>
            <div
              className="select__indicators css-1hb7zxy-IndicatorsContainer"
            >
              <span
                className="select__indicator-separator css-1okebmr-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                className="select__indicator select__dropdown-indicator css-tlfecz-indicatorContainer"
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
              >
                <i
                  aria-hidden="true"
                  className="fa fa-caret-down"
                />
              </div>
            </div>
          </div>
        </div>
        <input
          autoComplete="off"
          onChange={[Function]}
          required={true}
          style={
            Object {
              "height": 0,
              "opacity": 0,
              "position": "absolute",
              "width": "100%",
            }
          }
          tabIndex={-1}
          value={Array []}
        />
      </div>
    </div>
    <div
      className="formInput"
    >
      <label
        className="label required-label"
        htmlFor="overview"
      >
        OVERVIEW
      </label>
      <input
        autoComplete="off"
        className="input"
        defaultValue="Pulp Fiction is a 1994 American neo-noir black comedy crime film written and directed by Quentin Tarantino, who conceived it with Roger Avary.[4] Starring John Travolta, Samuel L. Jackson, Bruce Willis, Tim Roth, Ving Rhames, and Uma Thurman, it tells several stories of criminal Los Angeles. The title refers to the pulp magazines and hardboiled crime novels popular during the mid-20th century, known for their graphic violence and punchy dialogue."
        id="overview"
        name="overview"
        onChange={[Function]}
        placeholder="Overview here"
        required={true}
        type="text"
      />
    </div>
    <div
      className="formInput"
    >
      <label
        className="label"
        htmlFor="runtime"
      >
        RUNTIME
      </label>
      <input
        autoComplete="off"
        className="input"
        defaultValue={154}
        id="runtime"
        name="runtime"
        onChange={[Function]}
        placeholder="Runtime here"
        type="number"
      />
    </div>
    <div
      className="formInput"
    >
      <label
        className="label"
        htmlFor="budget"
      >
        BUDGET
      </label>
      <input
        autoComplete="off"
        className="input"
        defaultValue={55000000}
        id="budget"
        name="budget"
        onChange={[Function]}
        placeholder="Budget here"
        type="number"
      />
    </div>
    <div
      className="formInput"
    >
      <label
        className="label"
        htmlFor="revenue"
      >
        REVENUE
      </label>
      <input
        autoComplete="off"
        className="input"
        defaultValue={136906000}
        id="revenue"
        name="revenue"
        onChange={[Function]}
        placeholder="Revenue here"
        type="number"
      />
    </div>
    <div
      className="formInput"
    >
      <label
        className="label required-label"
        htmlFor="tagline"
      >
        TAGLINE
      </label>
      <input
        autoComplete="off"
        className="input"
        defaultValue="Mock tagline!"
        id="tagline"
        name="tagline"
        onChange={[Function]}
        placeholder="Tagline here"
        required={true}
        type="text"
      />
    </div>
    <div
      className="formInput"
    >
      <label
        className="label"
        htmlFor="vote_average"
      >
        VOTE AVERAGE
      </label>
      <input
        autoComplete="off"
        className="input"
        defaultValue={4.3}
        id="vote_average"
        name="vote_average"
        onChange={[Function]}
        placeholder="Vote average here"
        step="0.1"
        type="number"
      />
    </div>
    <div
      className="formInput"
    >
      <label
        className="label"
        htmlFor="vote_count"
      >
        VOTE COUNT
      </label>
      <input
        autoComplete="off"
        className="input"
        defaultValue={1195}
        id="vote_count"
        name="vote_count"
        onChange={[Function]}
        placeholder="Vote count here"
        type="number"
      />
    </div>
    <div
      className="modalFooter"
    >
      <button
        className="button"
        disabled=""
        id=""
        onClick={[Function]}
        type="reset"
      >
        RESET
        
      </button>
      <button
        className="button"
        disabled=""
        id=""
        onClick={[Function]}
        type="submit"
      >
        SAVE
        
      </button>
    </div>
  </form>,
]
`;
